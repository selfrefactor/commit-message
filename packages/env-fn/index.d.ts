type Modes = 'special' | 'local'

export function envFn(mode?: Modes, cwd?: string): string[]