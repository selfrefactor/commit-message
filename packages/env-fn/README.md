# Env-fn

Load .env configuration

## Usage

```
require("env-fn")()

require("env-fn")(".env")
require("env-fn")("special")
```

## Typing

`require("env-rn")(fileName: string)`

Searches for `fileName` starting from `process.cwd` directory and moving up.

If it finds it, it loads environment variables in the process.

Only the first request load variables; the following calls do nothing.

## Default value

- fileName - `.env`

## Pin to different directory

Default directory is `process.cwd()`
To apply local directory as start point use 'dirname' as argument, which is upgrade upon `special` behaviour.

## Forbid use through env flag

If your `process.env.ENV_FLAG` is different than `undefined` or `'true'`, then no environment variables will be loaded with this library.
