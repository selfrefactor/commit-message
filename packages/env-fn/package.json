{
	"dependencies": {
		"rambda": "6.4.0"
	},
	"name": "env-fn",
	"version": "2.2.0",
	"typings": "index.d.ts",
	"description": "Load environment variables from from local `.env` file",
	"main": "./src/env-fn.js",
	"repository": "https://github.com/selfrefactor/services/tree/master/packages/env-fn",
	"author": "Dejan Toteff <deyan8284@gmail.com>",
	"license": "MIT",
	"devDependencies": {}
}
