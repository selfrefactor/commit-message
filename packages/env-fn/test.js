const { envFn } = require('./envFn')

// SPECIAL
// ============================================
const envs = envFn('special')

//  LOCAL WITH CWD
// ============================================
// const envs = envFn('local', `${__dirname}/test_folder`)

// RESULTS
// ============================================
console.log('=============')
console.log({ envs })
console.log('=============')
console.log({
  redis : process.env.REDIS_URL,
  foo   : process.env.foo,
})
