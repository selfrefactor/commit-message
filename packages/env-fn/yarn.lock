# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1


rambda@6.4.0:
  version "6.4.0"
  resolved "https://registry.yarnpkg.com/rambda/-/rambda-6.4.0.tgz#bf080047b03d8d2c72e3ce5c17c603517a65749a"
  integrity sha512-aBoE7YrGQP1xnxjyg5fhupfS+FTrDVjZMtwpezl15RKLNnKLOYDfjx/0xDE+8GyWbbZfODRkAOZmnpvxrAONbQ==
