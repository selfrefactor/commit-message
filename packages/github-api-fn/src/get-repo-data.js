const { existsSync } = require('fs')
const { mapAsync, ok, pick } = require('rambdax')
const { outputJson, readJson } = require('fs-extra')

const { filterRepo, dateDiff } = require('./_modules/filter-repo')
const { getRepo } = require('./_modules/get-repo')
const cacheLocation = `${ __dirname }/cache.json`

async function getRepoData(input){
  ok(input)({ repos : [ String ] })
  const { repos, refreshCache = true } = input
  const reposOk = repos.filter(x => x.includes('/')).length === repos.length
  if (!reposOk) throw new Error(`Wrong repos input - ${ repos }`)

  if (refreshCache){
    const reposData = await mapAsync(async repo => {
      const repoDataResponse = await getRepo(repo)
      const filterData = await filterRepo(repo, input.daysLimit)
      const propsToPick = [
        'full_name',
        'description',
        'stargazers_count',
        'forks_count',
        'open_issues_count',
        'pushed_at',
        'updated_at',
        'subscribers_count',
      ]
      const repoData = pick(propsToPick, repoDataResponse)

      return {
        repoData,
        repoUrl : repo,
        filterData,
        pushedDiff: dateDiff(repoData.pushed_at),
        updatedDiff: dateDiff(repoData.updated_at),
      }
    }, repos)
    await outputJson(
      cacheLocation, { data : reposData }, { spaces : 2 }
    )

    return reposData
  }
  if (existsSync(cacheLocation)){
    const { data } = await readJson(cacheLocation)

    return data
  }
  throw new Error('cache is missing')
}

exports.getRepoData = getRepoData
