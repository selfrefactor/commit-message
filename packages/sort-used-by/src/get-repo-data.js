const { dropLast, filter, piped, remove, split } = require('rambdax')

function getRepoData(rawData){
  return piped(
    rawData,
    remove(/\n/g),
    split(' '),
    filter(x => x.trim()),
    x => ({
      repoUrl : dropLast(2, x).join(''),
      stars   : Number(x[ 3 ]),
    })
  )
}

exports.getRepoData = getRepoData
