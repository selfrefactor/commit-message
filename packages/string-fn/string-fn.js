export * from './src/between'
export * from './src/camelCase'
export * from './src/count'
export * from './src/constantCase'
export * from './src/distance'
export * from './src/distanceGerman'
export * from './src/dotCase'
export * from './src/glob'
export * from './src/getIndent'
export * from './src/indent'
export * from './src/isLetter'
export * from './src/isPunctuation'
export * from './src/kebabCase'
export * from './src/maskSentence'
export * from './src/maskWords'
export * from './src/ms'
export * from './src/pascalCase'
export * from './src/removeIndent'
export * from './src/randomString'
export * from './src/reverse'
export * from './src/seoTitle'
export * from './src/shuffle'
export * from './src/snakeCase'
export * from './src/splitPerLine'
export * from './src/splitSentence'
export * from './src/stripPunctuation'
export * from './src/getMaxLength'
export * from './src/fitWithinLines'
export * from './src/stripTags'
export * from './src/takeArguments'
export * from './src/titleCase'
export * from './src/trim'
export * from './src/words'
export * from './src/wordsX'
