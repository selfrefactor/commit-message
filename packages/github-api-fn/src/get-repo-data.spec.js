import { envFn } from 'env-fn'
import { ok } from 'rambdax'
import { ms } from 'string-fn'
envFn('special')
import { getRepoData } from './get-repo-data'
jest.setTimeout(ms('1min'))

const repos = [ 'selfrefactor/on', 'selfrefactor/rambdax' ]
const input = {
  repos,
  cacheLocation : `${ __dirname }/cache.json`,
  refreshCache  : true,
}

test('happy', async () => {
  const result = await getRepoData(input)
  ok(result)(Array)
  expect(
    result
  ).toMatchSnapshot()
})
