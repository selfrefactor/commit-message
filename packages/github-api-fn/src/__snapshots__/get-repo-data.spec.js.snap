// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy 1`] = `
Array [
  Object {
    "repoData": Object {
      "description": "An extremely fast JavaScript bundler and minifier",
      "forks_count": 356,
      "full_name": "evanw/esbuild",
      "open_issues_count": 86,
      "pushed_at": "2020-12-23T22:57:12Z",
      "stargazers_count": 16503,
      "subscribers_count": 208,
      "updated_at": "2020-12-26T17:37:55Z",
    },
    "repoUrl": "evanw/esbuild",
  },
  Object {
    "repoData": Object {
      "description": "JavaScript Regular expressions made easy",
      "forks_count": 521,
      "full_name": "VerbalExpressions/JSVerbalExpressions",
      "open_issues_count": 22,
      "pushed_at": "2020-12-18T22:05:17Z",
      "stargazers_count": 11764,
      "subscribers_count": 287,
      "updated_at": "2020-12-25T16:34:44Z",
    },
    "repoUrl": "VerbalExpressions/JSVerbalExpressions",
  },
  Object {
    "repoData": Object {
      "description": "Angular UI Component library for the Open Insurance Platform",
      "forks_count": 20,
      "full_name": "aposin/ng-aquila",
      "open_issues_count": 1,
      "pushed_at": "2020-12-21T19:51:04Z",
      "stargazers_count": 159,
      "subscribers_count": 16,
      "updated_at": "2020-12-23T19:11:02Z",
    },
    "repoUrl": "aposin/ng-aquila",
  },
  Object {
    "repoData": Object {
      "description": "Collection of wonderful TypeScript type gymnastics code snippets.",
      "forks_count": 4,
      "full_name": "g-plane/type-gymnastics",
      "open_issues_count": 0,
      "pushed_at": "2020-11-21T03:01:35Z",
      "stargazers_count": 77,
      "subscribers_count": 3,
      "updated_at": "2020-12-24T09:50:07Z",
    },
    "repoUrl": "g-plane/type-gymnastics",
  },
  Object {
    "repoData": Object {
      "description": "Refactor TypScript code programmatically using codemods",
      "forks_count": 0,
      "full_name": "WolkSoftware/tsmod",
      "open_issues_count": 1,
      "pushed_at": "2019-10-01T22:06:01Z",
      "stargazers_count": 68,
      "subscribers_count": 6,
      "updated_at": "2020-12-21T11:58:39Z",
    },
    "repoUrl": "WolkSoftware/tsmod",
  },
]
`;
